students/chelsea_nayan/lesson03/slicing_practice.py

```diff
@@ -0,0 +1,121 @@
+# chelsea_nayan, UWPCE, Python 210
+# Lesson 03: Slicing Lab Exercise
+
+# -------- MY TEST CASES -----------------
+seq_string1 = "This is a super cool string"
+seq_string2 = "s"
+seq_string3 = "0123456789-abcdefghijklmnopqrstuvwxyz"
+
+seq_tuple1 = (0, 1, "love", ["hi", "this", "me"], 2, "hey")
+seq_tuple2 = ("4",)
+seq_tuple3 = (0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17)
+
+seq_list1 = ["This", "is", "a", "list", "of", "strings", 1, 2, 3, 4, 5, 6, 7, 8]
+seq_list2 = [1]
+seq_list3 = ["Number", "time", 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, "end"]
+
+# ---------- THE FUNCTIONS ------------------
+
+# This function takes a sequence and returns a copy of that sequence with the first and last items exchanged
+def exchange_first_last(seq):
+    if len(seq) < 2: # This deals with sequences that have the length of 1. Returns the one item.
+        return seq
+    if type(seq) is list or tuple: # This deals with list or tuple type cases
+        new_seq = list(seq)
+        new_seq = [seq[-1]] + new_seq[1:-1] + [seq[0]] # Had to type cast the integers into lists by adding the [] around them. It was definitely hard figuring out how to fix the TypeError when I didn't have the [].
+    if type(seq) is str: # This deals with string type cases
+        new_seq = seq[-1] + seq[1:-1] + seq[0]
+    return type(seq)(new_seq) # Returns the new sequence and its correct type
+
+
+# This function takes a sequence and returns a copy of that sequence with every other item removed
+def rm_every_other(seq):
+    if type(seq) is list or tuple: # This deals with lists or tuples type cases
+        new_seq = list(seq)
+        new_seq = new_seq[::2] # Slices out every other item in the list or tuple
+    if type(seq) is str: # This deals with string type cases
+        new_seq = seq[::2] # Slices out every other item in the string
+    return type(seq)(new_seq) # Returns the new sequence and its correct type
+
+
+# This function takes a sequence and returns a copy of that sequence with the first 4 and the last 4 items removed, and then every other item inbetween starting from seq[4]
+def rm_ends_and_every_other(seq):
+    if type(seq) is list or tuple: # This deals with lists or tuples type cases
```

Comment:
incorrect logic, if you want to check if type is list or tuple:
`if isinstance(seq, (list, tuple))`
`if type(seq) is list or type(seq) is tuple`